src/script/components/MessagesList/index.tsx

```diff
@@ -17,47 +17,57 @@
  *
  */
 
-import React, {useEffect, useLayoutEffect, useRef, useState} from 'react';
+import cx from 'classnames';
+import React, {FC, MouseEvent as ReactMouseEvent, useEffect, useRef, useState} from 'react';
 import {ConversationRepository} from 'src/script/conversation/ConversationRepository';
 import {MessageRepository} from 'src/script/conversation/MessageRepository';
-import {Conversation} from '../../entity/Conversation';
+import {Conversation as ConversationEntity, Conversation} from '../../entity/Conversation';
 import {ContentMessage} from 'src/script/entity/message/ContentMessage';
 import {DecryptErrorMessage} from 'src/script/entity/message/DecryptErrorMessage';
 import {MemberMessage} from 'src/script/entity/message/MemberMessage';
 import {Message as MessageEntity} from 'src/script/entity/message/Message';
 import {User} from 'src/script/entity/User';
 import {StatusType} from '../../message/StatusType';
-import {registerReactComponent, useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
 import Message from './Message';
 import {Text} from 'src/script/entity/message/Text';
 import {useResizeObserver} from '../../ui/resizeObserver';
+import {isContentMessage} from '../../guards/Message';
+import {initFadingScrollbar} from '../../ui/fadingScrollbar';
 import {isMemberMessage} from '../../guards/Message';
 import {ServiceEntity} from 'src/script/integration/ServiceEntity';
+import onHitTopOrBottom from '../../ui/onHitTopOrBottom';
 
 type FocusedElement = {center?: boolean; element: Element};
+
 interface MessagesListParams {
   cancelConnectionRequest: (message: MemberMessage) => void;
   conversation: Conversation;
   conversationRepository: Pick<
     ConversationRepository,
-    'getPrecedingMessages' | 'getMessagesWithOffset' | 'updateParticipatingUserEntities' | 'expectReadReceipt'
+    | 'getPrecedingMessages'
+    | 'getMessagesWithOffset'
+    | 'updateParticipatingUserEntities'
+    | 'expectReadReceipt'
+    | 'getSubsequentMessages'
```

Comment:
Why to pick these methods here? I think we're always passing conversationRepository object as a whole 🤔